src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/SelectExpandNode.cs

```diff
@@ -32,6 +32,7 @@ public SelectExpandNode()
             SelectedStructuralProperties = new HashSet<IEdmStructuralProperty>();
             SelectedComplexProperties = new HashSet<IEdmStructuralProperty>();
             SelectedNavigationProperties = new HashSet<IEdmNavigationProperty>();
+            ExpandedNavigationPropertiesOnComplexTypes = new Dictionary<IEdmProperty, ExpandedNavigationSelectItem>();
             ExpandedProperties = new Dictionary<IEdmNavigationProperty, ExpandedNavigationSelectItem>();
```

Comment:
>ExpandedProperties [](start = 12, length = 18)

In V8, will we combine ExpandedNavigationPropertiesOnComplexTypes and ExpandedProperties?

Reply:
Maybe, I think there are some fundamental differences. The complex type dictionary can contain navigation select items even for it's nested properties where as ExpandedProperties only contains select items for navigation properties on the current structured type. If we are okay with that loss of information, we can probably merge these.

---
In reply to: [308487591](https://github.com/OData/WebApi/pull/1738#discussion_r308487591) [](ancestors = 308487591)